--- Config.pm.orig	2014-07-11 02:22:37.252961981 -0300
+++ Config.pm	2014-07-11 22:45:22.921191988 -0300
@@ -76,6 +76,40 @@
     # $Self->{SessionUseCookie} = 0;
     # $Self->{CheckMXRecord} = 0;
 
+    # Custom CustomerCompany for Support Quota Add-on     #
+    # --------------------------------------------------- #
+
+    $Self->{CustomerCompany} = {
+        Name   => 'Database Backend',
+        Module => 'Kernel::System::CustomerCompany::DB',
+        Params => {
+            Table => 'customer_company',
+            CaseSensitive => 0,
+        },
+
+        CustomerCompanyKey             => 'customer_id',
+        CustomerCompanyValid           => 'valid_id',
+        CustomerCompanyListFields      => [ 'customer_id', 'name' ],
+        CustomerCompanySearchFields    => ['customer_id', 'name'],
+        CustomerCompanySearchPrefix    => '',
+        CustomerCompanySearchSuffix    => '*',
+        CustomerCompanySearchListLimit => 250,
+        CacheTTL                       => 60 * 60 * 24, # use 0 to turn off cache
+
+        Map => [
+            [ 'CustomerID',             'CustomerID', 'customer_id', 0, 1, 'var', '', 0 ],
+            [ 'CustomerCompanyName',    'Customer',   'name',        1, 1, 'var', '', 0 ],
+            [ 'CustomerCompanyStreet',  'Street',     'street',      1, 0, 'var', '', 0 ],
+            [ 'CustomerCompanyZIP',     'Zip',        'zip',         1, 0, 'var', '', 0 ],
+            [ 'CustomerCompanyCity',    'City',       'city',        1, 0, 'var', '', 0 ],
+            [ 'CustomerCompanyCountry', 'Country',    'country',     1, 0, 'var', '', 0 ],
+            [ 'CustomerCompanyURL',     'URL',        'url',         1, 0, 'var', '$Data{"CustomerCompanyURL"}', 0 ],
+            [ 'CustomerCompanyComment', 'Comment',    'comments',    1, 0, 'var', '', 0 ],
+            [ 'CustomerCompanyQuota',   'Quota',      'quota',       1, 0, 'int', '', 0 ],
+            [ 'ValidID',                'Valid',      'valid_id',    0, 1, 'int', '', 0 ],
+        ],
+    };
+
     # ---------------------------------------------------- #
 
     # ---------------------------------------------------- #
